### ISSUE TITLE
Attack time can't be modulated when it's set to 0

### ISSUE THREAD

**Issue by TheNerdyMusicGuy** [surge-synthesizer/surge#2650]
**Describe the bug**
When the attack time of any of the envelopes is set to 0, it can not be modulated. This also appears to apply to other envelope parameters, like the decay time, but with those it only starts working with a value higher than ~0.009.

**Please let us know your surge version**
 - Surge Version: Surge-NIGHTLY-2020-09-19-4d8baf1 (this bug also happened in earlier versions)
 - Plugin type: VST3
 - Bits: 64

**To Reproduce**
1. Start with the Init patch.
2. Assign velocity to the attack time of the amp envelope and set some modulation amount.
3. A high velocity should now result in a higher attack time, but it doesn't.
4. Raise the attack time slightly, to any other value than 0.
5. Now the modulation applies.

**Desktop (please complete the following information):**
 - OS: Windows 8.1
 - Host: Ableton Live 10

**Additional context**
@baconpaul Suggested this might have to do something with the modulated parameter being lover than the block size.

**Comment by baconpaul** [comment]
yeah thesis is: if attack is < block size and the modulation is out of order / accidentally lagged by one block then the attack will be 'over' before the velocity gets there.

Same thing happens with a square wave LFO applied.

**Comment by baconpaul** [comment]
OK yeah so that' exactly the bug

If in surge voice I move the ameg->process_block to behind where the modulation is applied, then voila the ampeg receives the modulation

but then,  of course, you end up with the problem that the ampeg doesn't participate in the modulation properly

hmm i need to think about this one. In that the envelopes are stateful and can 'cross over' the attack before they get the information they are special.

**Comment by baconpaul** [comment]
So: I don't see an easy way to fix this. The problem is the envelope runs and if the time is exactly 0, it swaps state (basically going immediately to delay or if A and D are zero, immediately to S) and that state doesn't 'back down' if attack modulates.

The correct change I think is to check in the decay state if you are earlier than current attack (so you were modulated) but then do what exactly?

And "some" modulator has to run first in the current architecture. Doing a topological sort of modulators would help but there can be cycles.

So what I'm going to do is push this out of 1.8; I'll suggest that @VincyZed add a comment in the manual that "if ADR are zero, they will not respond to modulation". And then in 1.9 I continue to ponder. But there's no small fix, there's an obvious workaround, and we are late in the cycle.
